src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -216,17 +218,35 @@ def _create_task_list(source_data_dir, kbase_collection, load_ver, tool, wrapper
     return task_list_file, len(genome_ids_chunks)
 
 
-def _create_batch_script(job_dir, task_list_file, n_jobs):
+def _cal_node_num(tool, n_jobs):
+    """
+    Calculate the number of nodes required for the task
+    """
+
+    tool_exe_time = TASK_META[tool]['exe_time']
+    jobs_per_node = NODE_TIME_LIMIT * 60 // tool_exe_time
+
+    num_nodes = min(math.ceil(n_jobs / jobs_per_node), MAX_NODE_NUM)
```

Comment:
What happens to the jobs that can't run because there aren't enough nodehours scheduled if `n_jobs/jobs_per_node > MAX_NODE_NUM`?

Reply:
Those nodes will hit wall-time and get killed. And re-submit is needed. Do you think we should remove `MAX_NODE_NUM`?

Reply:
My preference would be to not allow users to schedule a set of jobs where some are guaranteed to fail. Right now that means they're requesting 10000 hours of compute, which for our case, seems like plenty given running GTDB_TK on GTDB, which is the most expensive compute we currently support, is only ~500 hours.

Reply:
👍  updated MAX_NODE_NUM = 100, and raise an error if the requested node num is greater than it, meaning they are requesting more than 1000 total node compute time.

Reply:
:+1: